lib/runners/cloud-hypervisor.nix

```diff
@@ -30,6 +37,24 @@ let
 
   kernelCmdLine = "${kernelConsole} reboot=t panic=-1 ${builtins.unsafeDiscardStringContext (toString microvmConfig.kernelParams)}";
 
+
+  userVSockOpts = (extractOptValues "--vsock" extraArgs).values;
```

Comment:
Why do it like this instead of getting these values from `microvmConfig`?

Reply:
I'm happy to implement all the chv block options under microvm.cloud-hypervisor.vsock of you want. I don't think it's worth it though. I only added the --vsock parsing here because I think escape hatches are important. `extraArgs` is fundamentally an escape hatch. How do you want to proceed?